xbmc/platform/win32/WinMain.cpp

```diff
@@ -38,7 +38,28 @@ LONG WINAPI CreateMiniDump(EXCEPTION_POINTERS* pEp)
 //-----------------------------------------------------------------------------
 INT WINAPI WinMain(HINSTANCE hInst, HINSTANCE, LPSTR commandLine, INT)
 {
+  bool portable = false;
+  int argc = 0;
+  LPWSTR* argvW = CommandLineToArgvW(GetCommandLineW(), &argc);
+
+  char** argv = new char*[argc];
+
+  for (int i = 0; i < argc; ++i)
+  {
+    int size = WideCharToMultiByte(CP_UTF8, 0, argvW[i], -1, nullptr, 0, nullptr, nullptr);
+    if (size > 0)
+    {
+      argv[i] = new char[size];
+      WideCharToMultiByte(CP_UTF8, 0, argvW[i], -1, argv[i], size, nullptr, nullptr);
+
+      std::string param(argv[i]);
+      if (param == "-p" || param == "--portable")
```

Comment:
This shouldn't be needed, arguments are parsed uniformly by all platforms in https://github.com/xbmc/xbmc/blob/master/xbmc/AppParamParser.cpp
If you need the result earlier maybe you could move the call to AppParamParser::Parse

Reply:
I had done it like this to avoid change more things in WinMain but I agree that may be better use `AppParamParser` for all.

Updated.